mamba/mamba/utils.py

```diff
@@ -102,10 +102,23 @@ def fixup_channel_spec(spec):
                 channel, channel_platform, full_url, pkgs_dirs, repodata_fn
             )
 
+            needs_finalising = sd.download_and_check_targets(dlist)
             index.append(
-                (sd, {"platform": channel_platform, "url": url, "channel": channel})
+                (
+                    sd,
+                    {
+                        "platform": channel_platform,
+                        "url": url,
+                        "channel": channel,
+                        "needs_finalising": needs_finalising,
+                    },
+                )
             )
-            dlist.add(sd)
+
+    for (sd, info) in index:
+        if info["needs_finalising"]:
+            sd.finalize_checks()
```

Comment:
I'm not too familiar with this part of the code base, can you explain what is going on here? Are there any alternatives to having this logic here in Python?

Reply:
This is effectively just me mimicking the code that's in [`channel_loader`](https://github.com/mamba-org/mamba/blob/15f6b6e6062ce33a4562ef7003b90c42adfd329a/libmamba/src/api/channel_loader.cpp#L111-L119) which atm is duplicated between mamba and micromamba.

Let me know if you have an idea in mind that I could test out to make this nicer.

Reply:
More concretely though, what's happening in the zst-enabled codepaths is that we do a HEAD request first to check if the zst variant exists, and if it does, we add another target that's the actual GET request for the zst variant.